frontend/components/details/ModuleHTML.tsx

```diff
@@ -0,0 +1,60 @@
+import { HTMLReactParserOptions } from "html-react-parser";
+import { Element } from "html-react-parser/node_modules/domhandler/lib/node";
+import { FC, ReactElement, useEffect, useState } from "react";
+
+interface ModuleHTMLProps {
+  html: string;
+}
+
+const ModuleHTML: FC<ModuleHTMLProps> = ({ html }: ModuleHTMLProps) => {
+  const [renderedHTML, setRenderedHTML] = useState(() => (
+    <div dangerouslySetInnerHTML={{ __html: html }} />
+  ));
+
+  useEffect(() => {
+    Promise.all([import("html-react-parser"), import("@/components/details/ModuleLink")]).then(
+      ([{ default: parse, domToReact, attributesToProps }, { default: ModuleLink }]) => {
+        // const  = module;
+        const options: HTMLReactParserOptions = {
+          // eslint-disable-next-line react/display-name
+          replace: (domNode) => {
+            if (domNode.type !== "tag") {
+              return;
+            }
+            const element = domNode as Element;
```

Comment:
The cast here is necessary to get around a weird bug when using this package with NextJS and TS.
https://github.com/remarkablemark/html-react-parser/issues/221